core/vm/interpreter.go

```diff
@@ -241,6 +249,11 @@ func (in *EVMInterpreter) Run(contract *Contract, input []byte, readOnly bool) (
 				return nil, ErrWriteProtection
 			}
 		}
+		if in.indestructible {
```

Comment:
This is very similar functionality to the check that `opSelfdestruct` does. Is it possible to rely only on one or the other? Is this statement catching a particular code path? If possible - I think it is best to rely on the implementation of this in `opSelfdestruct` so each opcode doesn't need to have a `operation.selfdestructs` flag.

Reply:
You're right. The operation.selfdestructs flag is unnecessary.